src/cgame/cg_gameinfo.cpp

```diff
@@ -31,12 +31,234 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+			  example: "sarge, uriel",
+			  used by Quake 3, Q3 Rally…
+			* bot_skill: bot skill level,
+			  example: 3,
+			  used by Quake Live.
+			* timelimit: timelimit in minutes,
+			  example: 10,
+			  used by Quake 3, Quake Live, Wild West,
+			  Return to Castle Wolfenstein,
+			  Wolfenstein: Enemy Territory…
+			* fraglimit: amount of frags to end a game,
+			  example: 20.
+			  used by Quake 3, Quake Live…
+			* briefing: a string displayed to the player
+			  before he enters the match,
+			  used by Wolfenstein: Enemy Territory.
+			* desc_line1 (2, 3 ,4): lines of text describing
+			  the map,
+			  used by Smokin' Guns.
+
+			And others.
+
+			They are ignored for now as we don't make use of them.
+
+			As this code does not make use of “Info” storage format,
+			we can also implement list of things by using multiple
+			time the same key name, like the ogg metadata format
+			does.
+
+			Example, instead of:
+
+			  type "tourney ffa team"
+
+			We can implement:
+
+			  type "tourney"
+			  type "ffa"
+			  type" team"
+
+			This is also similar to how it is done in the .mapinfo
+			files from Xonotic (using a different file format). */
+		}
+
+		if ( !arenaInfo.name.empty() )
+		{
+			arenaList.push_back( arenaInfo );
+		}
+	}
+}
+
+/*
+===============
+CG_LoadArenasFromFile
+===============
+*/
+static bool CG_LoadArenasFromFile( std::string filename, bool legacy = false )
+{
+	if ( legacy )
+	{
+		Log::Debug( "Loading legacy arena file: %s", filename );
+	}
+	else
+	{
+		Log::Debug( "Loading arena file: %s", filename );
+	}
+
+	std::error_code err;
+	std::string text = FS::PakPath::ReadFile( filename, err );
+
+	if ( err )
+	{
+		if ( !legacy )
+		{
+			Log::Warn( "Arena file not found: %s", filename );
+		}
+
+		return false;
+	}
+
+	CG_ParseInfos( text.c_str() );
+
+	return true;
+}
+
 /*
 ===============
 CG_LoadArenas
 ===============
 */
-void CG_LoadArenas()
+void CG_LoadArenas( std::string mapname )
+{
+	arenaList.clear();
+
+	std::string filename = Str::Format( "meta/%s/%s.arena", mapname, mapname );
+
+	if ( !CG_LoadArenasFromFile( filename ) )
+	{
+		/* Legacy games may store more than one map metadata in a
+		single .arena file, the .arena file not having a basename
+		in common with any map.
+
+		The .arena files (whatever the amount of map described in)
+		may be stored in various folders. Many games use the scripts/
+		folder, some may use the maps/ one (like Smokin' Guns).
+
+		Some games like Quake 3 and Quake Live also reads a file
+		named scripts/arenas.txt and Quake Live also reads files
+		with .sp_arena extension in script/ for single-player
+		training maps. */
+
+		char dirlist[ 4096 ];
+		int numdirs = trap_FS_GetFileListRecursive( "", ".arena", dirlist, sizeof( dirlist ) );
+		char *dirptr = dirlist;
+
+		for ( int i = 0, dirlen = 0; i < numdirs; i++, dirptr += dirlen + 1 )
+		{
+			dirlen = strlen( dirptr );
+
+			Log::Warn( "Loading legacy arena file: %s", dirptr );
```

Comment:
probably shouldn't be a warning

Reply:
This line was even a duplicate of a Debug print elsewhere. Now removed.